site/_posts/2022-11-01-authenticators-ceremonies-webauthn-oh-my.md

```diff
@@ -0,0 +1,128 @@
+* It's easier for attackers to steal a user's password than to steal their device. Even if an attacker does steal a device, they still need to bypass the device's security (*ie* biometrics or facial recognition system).
+* WebAuthn is easy to implement because it consists of [browser APIs that can be easily used by developers](https://developer.mozilla.org/en-US/docs/Web/API/Web_Authentication_API#examples).
+* It offers a smooth authentication experience for users.
+
+## How WebAuthn Works
+
+WebAuthn consists of three components: the authenticator (the user's device), the web browser, and the server (RP). In essence, the user initiates login via the browser, the browser then prompts the user to authenticate using their device (via biometrics or facial recognition, for example), and then if successful, the user is logged into the web application.
+
+However, this is only a basic view of the process. To understand the process in more depth, you need to look at the two main flows of WebAuthn: the registration flow and the authentication flow. (They are called "ceremonies" in the specification.)
+
+### The Registration Flow
+
+Before users can authenticate, their device(s) must be registered on the server of the RP. This way, the RP creates the credentials that the user can authenticate, and the process looks like this:
+
+{% include _image.liquid src="/assets/img/blogs/webauthn-terminology/registration-flow.png" alt="WebAuth Registration Flow diagram" class="img-fluid" figure=true %}
+
+1. The user visits the web application of the RP and initiates registration.
+2. The server creates a challenge key for registration, which is only used once.
+3. The challenge key, along with the user's info, is sent to the user's device.
+4. The user is prompted to authenticate themselves (this could be via biometrics, facial recognition system, or other forms depending on the device).
+5. After authentication success, the authenticator (the user's device) creates a new key pair (the public and private keys). 
+6. The authenticator uses the private key to sign the challenge and then sends the public key and a credential identifier alongside the challenge over to the server.
+7. The server validates the credibility of the private key by using the public key to evaluate the signed challenge.
+8. The server associates the credentials with the user's account for future authentication purposes and redirects them to the application.
+
+### The Authentication Flow
+
+After a user's device is registered with the RP, the user can be authenticated. There are six steps involved in the WebAuthn authentication process:
+
+{% include _image.liquid src="/assets/img/blogs/webauthn-terminology/authentication-flow.png" alt="WebAuth Authentication Flow diagram" class="img-fluid" figure=true %}
```

Comment:
I love the visuals, but communication with the authenticator is mediated by the browser. Readers may misinterpret the diagrams as meaning that the server communicates directly with the authenticator during these ceremonies.

Maybe rather than changing the visuals it's enough to make a note on the steps below like:
> The server (RP) creates and sends a unique challenge to the authenticator (via the browser).

Or perhaps it's clear to others that all of the interaction with the RP server goes through the browser because that's how you're browsing.